src/expandableCalendar/WeekCalendar/index.tsx

```diff
@@ -50,9 +50,23 @@ const WeekCalendar = (props: WeekCalendarProps) => {
   const list = useRef<FlatList>(null);
   const currentIndex = useRef(NUMBER_OF_PAGES);
 
+  useDidUpdate(() => {
+    items.current = getDatesArray(date, firstDay, numberOfDays);
+    setListData(items.current);
+    visibleWeek.current = date;
+  }, [numberOfDays]);
+
   useDidUpdate(() => {
     if (updateSource !== UpdateSources.WEEK_SCROLL) {
-      const pageIndex = items.current.findIndex(item => sameWeek(item, date, firstDay));
+      const pageIndex = items.current.findIndex(
+        item => (numberOfDays && numberOfDays > 1) ?
```

Comment:
Better move this condition (numberOfDays && numberOfDays > 1) to a const since you're using the same one later in line 250